### PR TITLE
cweb: init at 22p

### PR DESCRIPTION
**Pull request by vrthra**
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/misc/tie/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+stdenv.mkDerivation rec {
+  name = "tie-${version}";
+  version = "2.4";
+
+  src = fetchurl {
+    url = "http://mirrors.ctan.org/web/tie/${name}.tar.gz";
+    sha256 = "1m5952kdfffiz33p1jw0wv7dh272mmw28mpxw9v7lkb352zv4xsj";
+  };
+
+  buildInputs = [ ];
```
Comment (joachifm) [inline pkgs/development/tools/misc/tie/default.nix]: This is redundant

File: pkgs/development/tools/misc/tie/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchurl, coreutils }:
```
Comment (joachifm) [inline pkgs/development/tools/misc/tie/default.nix]: coreutils appears unused

File: pkgs/development/tools/misc/cwebbin/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    rev = name;
+    sha256 = "0zf93016hm9i74i2v384rwzcw16y3hg5vc2mibzkx1rzvqa50yfr";
+  };
+
+  cweb = fetchurl {
+    url = https://www.ctan.org/tex-archive/web/c_cpp/cweb/cweb.tar.gz;
+    sha256 = "1hdzxfzaibnjxjzgp6d2zay8nsarnfy9hfq55hz1bxzzl23n35aj";
+  };
+
+  buildInputs = [ coreutils tie ];
```
Comment (joachifm) [inline pkgs/development/tools/misc/cwebbin/default.nix]: Why is coreutils added here? They should be in the standard build environment.

Comment (vrthra) [inline pkgs/development/tools/misc/cwebbin/default.nix]: @joachifm fixed

File: pkgs/development/tools/misc/tie/default.nix (hunk 2 above)
Comment (vrthra) [inline pkgs/development/tools/misc/tie/default.nix]: @joachifm fixed

File: pkgs/development/tools/misc/tie/default.nix (hunk 1 above)
Comment (vrthra) [inline pkgs/development/tools/misc/tie/default.nix]: @joachifm removed

File: pkgs/development/tools/misc/cwebbin/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  buildInputs = [ tie ];
+
+  buildPhase = ''
+    sed -ie 's#/usr/local#$(out)#g' Makefile*
+    sed -ie 's#/usr#$(out)#g' Makefile*
+    zcat ${cweb} | tar -xvpf -
+    make -f Makefile.unix boot
+    make -f Makefile.unix cautiously CP=cp RM=rm DESTDIR=$out/bin MANDIR=$out/share/man/man1 DESTPREF=c
```
Comment (joachifm) [inline pkgs/development/tools/misc/cwebbin/default.nix]: What about factoring out make arguments used both here and under installation into `makeFlags`?

Comment (vrthra) [inline pkgs/development/tools/misc/cwebbin/default.nix]: @joachifm updated.

Comment (vrthra) [inline pkgs/development/tools/misc/cwebbin/default.nix]: @joachifm fixed

**Pull request merged**